contracts/AelinPool.sol

```diff
@@ -141,6 +141,17 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
                             NftCheck.supports721(nftCollectionRules[i].collectionAddress),
                         "can only contain 721"
                     );
+
+                    uint256 rangesLength = nftCollectionRules[i].idRanges.length;
+                    require(rangesLength <= 10, "max of ten id ranges");
```

Comment:
maybe let's put `10` into a constant and error should be simpler like "max ranges reached" or "too many ranges"